### PR TITLE
Initial repo reorg change

### PR DESCRIPTION
**Pull request by kvmware**
New READMEs and some migrated info from older ones. Below is proposed layout:

prow/
|- cluster-definitions/
|---- control-plane
|-------- 100-700-*.yaml (move from .)
|---- boskos*.yaml (remove parent folder, keep just the single yaml. This maybe should be moved this to the gcp/boskos folder)
|---- build/
|---- monitoring/
|---- prowjob-crd*.yml (remove parent folder)
|---- Makefile* (moved from ../)
|- job-definitions/ (renamed from jobs_config)
|- jobs/
|- config/
|---- prow-components.yaml (renamed from config.yaml)
|---- hook-plugins.yaml (renamed from plugins.yaml)
|---- labels.yaml (moved from config/label_sync)
|---- testgrid/
|-------- *-testgrid-*.yaml (moved from config/prow/k8s-testgrid)

gcp/
|- (move all infra/gcp files here)

### CODE REVIEW

**Comment by chizhg** [comment]
There are a few issues with this:
1. `boskos.yaml` is not a Kubernetes resource config file, so it cannot be created with `kubectl apply` thus should not be under the same folder as other Kubernetes resource config files.
2. It's better to not rename `config.yaml` and `plugins.yaml` since they were hardcoded somewhere in Prow IIRC, also these are the names used by all Prow instances I've seen.
3. `labels.yaml` is not part of Prow so it should not be under the Prow config folder.
4. For separation of concerns, I would prefer not having the `Makefile` in the same folder of the yaml files.

My suggestion of the new layout is like below:

prow/
|- cluster-definitions/
|---- control-plane
|-------- 100-700-.yaml
|-------- prowjob-crd*.yaml
|---- build/
|-------- 100-600-.yaml
|-------- boskos-resources
|------------- boskos_resources.yaml
|---- monitoring/
|---- Makefile*
|- job-definitions/ (renamed from jobs_config)
|- jobs/
|- config/
|---- config.yaml
|---- plugins.yaml
|---- testgrid/
|-------- -testgrid-.yaml (moved from config/prow/k8s-testgrid)

label_sync/
|---- labels.yaml

gcp/
|- (move all infra/gcp files here)

**Comment by kvmware** [comment]
1. That make sense that you don't want the boskos.yaml with other, the initial proposal does not have it mixed with any of the other yaml. For the most part one would `kubectl apply` a directory in the case of the proposed layout would be `cluster-definitions/control-plane/` I am fine with changing `control-plane` to core also. However, the point is that boskos.yaml will be in the `cluster-definitions/` folder with no other yaml files, thus shouldn't be the problem you are worried about no?
2. Those file names are configurable https://github.com/knative/test-infra/search?q=plugins.yaml. However, it is not a big win just makes things easier to understand, but that can be accomplished with the explanation of what the files are in the README.md which I have done.
3. Ah. It is a prow job, but yea I am fine with it not being under prow/
4. I put it under `cluster-definitions` as usually the Makefiles are in places where they do the action, but no strong feelings there.

In summary, I guess the boskos.yaml position is the only thing that I don't like. It is separated from the other yaml via the `core`|`control-plane` and `build` directories and is not mixed with any other yaml, we can put it in a `boskos` directory, but it seems unnecessary since it is just one file. Is there any strong reason for it being buried under the `build` directory?

**Comment by kvmware** [comment]
/hold
I have to make changes to the READMEs to reflect the changes. Will remove the hold once we come to a conclusion.

**Comment by kvmware** [comment]
As per our out of bound conversation Chi:
move the label-sync to community
go forward with your layout

**Comment by kvmware** [comment]
Made the changes, should be ready for final pass @chizhg @upodroid

**Review by chizhg — COMMENTED** [review]
Without reading the READMEs in detail yet (I will do another pass of review later today) - given it's Knative release week, shall we only copy the files now, and only make job switches and cleanups for the test-infra repo after the release? WDYT?

**Comment by kvmware** [comment]
> Without reading the READMEs in detail yet (I will do another pass of review later today) - given it's Knative release week, shall we only copy the files now, and only make job switches and cleanups for the test-infra repo after the release? WDYT?

Yea, this PR only has READMEs but i al think we should just do a copy of the files and don't initiate any job switches until we are ready.

**Comment by kvmware** [comment]
all MDs linted

**Review by upodroid — APPROVED** [review]
LGTM

I can add additional details when I move infra/gcp from test-infra
